test/benchmarks/benchmark_compare258.cc

```diff
@@ -0,0 +1,90 @@
+/* benchmark_compare258.cc -- benchmark compare258 variants
+ * Copyright (C) 2020 Nathan Moinvaziri
+ * For conditions of distribution and use, see copyright notice in zlib.h
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdint.h>
+
+#include <benchmark/benchmark.h>
+
+extern "C" {
+#  include "zbuild.h"
+#  include "zutil.h"
+#  include "cpu_features.h"
+}
+
+#define MAX_COMPARE_SIZE (258)
+
+typedef uint32_t (*compare258_func)(const unsigned char *src0, const unsigned char *src1);
+
+class compare258: public benchmark::Fixture {
+private:
+    uint8_t str1[274];
+    uint8_t str2[274];
+public:
```

Comment:
Minor suggestion:
```
diff --git a/test/benchmarks/benchmark_compare258.cc b/test/benchmarks/benchmark_compare258.cc
index bd75120..6b11088 100644
--- a/test/benchmarks/benchmark_compare258.cc
+++ b/test/benchmarks/benchmark_compare258.cc
@@ -21,8 +21,8 @@ typedef uint32_t (*compare258_func)(const unsigned char *src0, const unsigned ch
 
 class compare258: public benchmark::Fixture {
 private:
-    uint8_t str1[274];
-    uint8_t str2[274];
+    uint8_t str1[MAX_COMPARE_SIZE];
+    uint8_t str2[MAX_COMPARE_SIZE];
 public:
     void SetUp(const ::benchmark::State& state) {
         memset(str1, 'a', sizeof(str1));
@@ -33,7 +33,7 @@ public:
         int32_t x = (int32_t)state.range(0);
         uint32_t len;
 
-        while (state.KeepRunning()) {
+        for (auto _ : state)
             str2[x] = 0;
             len = compare258((const uint8_t *)str1, (const uint8_t *)str2);
             str2[x] = 'a';
```

Unless you were oversizing those strings on purpose for alignment purposes?

Reply:
Err, the diff is missing a brace, but you get the gist.

Reply:
I think there was a reason I was oversizing them but I can't remember now. Anyways I have fixed it.